pkg/syncer/spec/spec_process.go

```diff
@@ -351,10 +352,10 @@ func transformName(syncedObject *unstructured.Unstructured) {
 	if syncedObject.GroupVersionKind() == configMapGVR && syncedObject.GetName() == "kube-root-ca.crt" {
 		syncedObject.SetName("kcp-root-ca.crt")
 	}
-	// TODO(jmprusi): We are rewriting the name of the object into a non random one so we can reference it from the deployment transformer
-	//                but this means that means than more than one default-token-XXXX object will overwrite the same "kcp-default-token"
-	//				  object. This must be fixed.
-	if syncedObject.GroupVersionKind() == secretGVR && strings.Contains(syncedObject.GetName(), "default-token-") {
-		syncedObject.SetName("kcp-default-token")
+	if syncedObject.GroupVersionKind() == secretGVR && strings.Contains(syncedObject.GetName(), "-token-") {
```

Comment:
Let's remove the check for `-token-`. If it's a service account secret, it will have the annotation - just check for that.

Reply:
done